crates/db_schema/src/impls/instance.rs

```diff
@@ -249,3 +249,53 @@ impl Blockable for InstanceActions {
       .await
   }
 }
+
+impl InstanceActions {
+  pub async fn check_ban(
+    pool: &mut DbPool<'_>,
+    person_id: PersonId,
+    instance_id: InstanceId,
+  ) -> LemmyResult<()> {
+    let conn = &mut get_conn(pool).await?;
+    let instance_actions = instance_actions::table
+      .filter(instance_actions::person_id.eq(person_id))
+      .filter(instance_actions::instance_id.eq(instance_id))
+      .order_by(instance_actions::blocked)
+      .get_result::<Self>(conn)
+      .await
+      .optional()?;
+
+    let received_ban = instance_actions.iter().any(|i| i.received_ban.is_some());
+    if received_ban {
+      return Err(LemmyErrorType::SiteBan.into());
+    }
+    Ok(())
+  }
```

Comment:
The `crates/db_schema/src/impls/community.rs` `read_blocks` function has a better example of how to do this.

It'd also be a good idea to add a helper function to `check_ban_local_or_home` .

Reply:
Its using an iter, not using select / exists / then_some...

Reply:
Right, fixed